src/autoTune.ml

```diff
@@ -126,55 +139,81 @@ let addModAttributes file =
 
 
 let disableIntervalContextsInRecursiveFunctions () =
-  ResettableLazy.force functionCallMaps |> fun (x,_,_) -> x |> FunctionCallMap.iter (fun f set ->
+  (ResettableLazy.force functionCallMaps).calling |> FunctionCallMap.iter (fun f set ->
       (*detect direct recursion and recursion with one indirection*)
       if FunctionSet.mem f set || (not @@ FunctionSet.disjoint (calledFunctions f) (callingFunctions f)) then (
         print_endline ("function " ^ (f.vname) ^" is recursive, disable interval and interval_set contexts");
         f.vattr <- addAttributes (f.vattr) [Attr ("goblint_context",[AStr "base.no-interval"; AStr "base.no-interval_set"; AStr "relation.no-context"])];
       )
     )
 
+let hasFunction pred =
+  let relevant_static var =
+    if LibraryFunctions.is_special var then
+      let desc = LibraryFunctions.find var in
+      GobOption.exists (fun args -> pred (desc.special args)) (functionArgs var)
+    else
+      false
+  in
+  let relevant_dynamic var =
+    if LibraryFunctions.is_special var then
+      let desc = LibraryFunctions.find var in
+      (* We don't really have arguments at hand, so we cheat and just feed it a list of Cil.one of appropriate length *)
+      match unrollType var.vtype with
+      | TFun (_, args, _, _) ->
+        let args = BatOption.map_default (List.map (fun (x,_,_) -> Cil.one)) [] args in
```

Comment:
`MyCFG.unknown_exp` might be semantically more appropriate although it doesn't make a difference to the classification.